src/Bicep.Core.UnitTests/Diagnostics/LinterRuleTests/ArtifactsParametersRuleTests.cs

```diff
@@ -0,0 +1,498 @@
+                    // For Bicep we don't care about that, but still both parameters must be present if either of them are.
+                    "If an '_artifactsLocation' parameter is provided, an '_artifactsLocationSasToken' parameter must also be provided."
+                }
+           );
+        }
+
+        [TestMethod]
+        public void ArmTtk_MissingLocation_Fail()
+        {
+            CompileAndTest(
+                @"
+                   param _artifactsLocationSasToken string
+                ",
+                new string[]
+                {
+                    // In the ARM TTK for a main template, this would have failed for this reason:
+                    //   The _artifactsLocationSasToken in "mainTemplate.json" has an incorrect defaultValue, must be an empty string
+                    // For Bicep we don't care about that, but still both parameters must be present if either of them are.
+                    "If an '_artifactsLocationSasToken' parameter is provided, an '_artifactsLocation' parameter must also be provided."
+                }
+           );
+        }
+
+        [TestMethod]
+        public void ArmTtk_WrongDefaults_Fail()
+        {
+            /* TTK result:
+              [-] artifacts parameter (3 ms)                                                                                      
+                    ENV:SAMPLE_NAME is empty - using placeholder for manual verification: 100-blank-template                        
+                    The _artifactsLocation in ""mainTemplate.json"" has an incorrect defaultValue, found: http://myweb.com/           
+                    Must be one of: https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/100-blank-template/ or deployment().properties.templateLink.uri 
+            */
+
+            /*                                                                 
+                [-] Secure String Parameters Cannot Have Default (3 ms)                                                             
+                    Parameter _artifactsLocationSasToken is a SecureString and must not have a default value unless it is an expression that contains the newGuid() function.
+             */
+            CompileAndTest(
+                @"
+                    param _artifactsLocation string = 'something besides an http'
+
+                    @secure()
+                    param _artifactsLocationSasToken string = 'hard coded token' //this shouldn't be a default (use a param if needed)
+                ",
+                new string[]
+                {
+                    $"If the '_artifactsLocation' parameter has a default value, it must be a raw URL or an expression like 'deployment().properties.templateLink.uri'.",
+                    $"If the '_artifactsLocationSasToken' parameter has a default value, it must be an empty string.",
+                });
+        }
+
+        [TestMethod]
+        public void WrongDefaults2_Pass()
```

Comment:
This test looks similar to ArmTtk_RawRepoPath_Pass, except the other one has @secure for both params. Why is it called
WrongDefaults2_Pass? The defaults look acceptable.

Reply:
Removed